### ISSUE TITLE
The provider does not work on .NET Core

### ISSUE THREAD

**Issue by psfinaki** [fsprojects/AzureStorageTypeProvider#111]
I was struggling with extracting data from my Azure Table Storage using the .NET API and then discovered this wonderful type provider - thanks for creating it!

Now, the thing is that my app runs on .NET Core and this provider does not seem to work properly with that, at least out of the box.

I have created basic console apps for .NET Framework and .NET Core, installed there the [nuget](https://www.nuget.org/packages/FSharp.Azure.StorageTypeProvider) and added the following code:
```fsharp
open FSharp.Azure.StorageTypeProvider

type Azure = AzureTypeProvider<"proper connection string">

[<EntryPoint>]
let main argv = 
    let entities = Azure.Tables.TestTable.Query().Execute()
    0
```
.NET Framework app does the job, .NET Core app throws `MissingMethodException`:
```
System.MissingMethodException: 'Method not found: 'System.Collections.Generic.IEnumerable`1<!!0> 
Microsoft.WindowsAzure.Storage.Table.CloudTable.ExecuteQuery(Microsoft.WindowsAzure.Storage.Table.TableQuery`1<!!0>, 
Microsoft.WindowsAzure.Storage.Table.TableRequestOptions, Microsoft.WindowsAzure.Storage.OperationContext)'.'
```
Can I do something about it? Maybe add some binding redirect somewhere? Or some blacker magic?

**Comment by isaacabraham** [comment]
@psfinaki I've got some bad news - currently there's no support for .NET Core. This basically stems from three things:

1. From what I understand, Type Providers on .NET Core don't yet support resolution of calls to third-party assemblies. In this case, this refers to the actual Azure Storage SDK, which the type provider "wraps around" to do all the heavy lifting. This is a blocker at the moment for porting to .NET Core.

2. .NET Core-compatible versions of the Azure Storage SDK have removed many calls from the API that the non-.NET Core SDK had (mostly around non-async methods). Why they elected to do this I don't know but it will mean rewriting large chunks of the TP to support them.

3. I've considered doing a "bare bones" TP that goes straight to the Azure REST API, avoiding dependencies completely, but this would be essentially a complete rewrite, but the need to find a file-based embedded JSON-to-F# deserializer.

So in summary - I also want a .NET Core version of the TP, but at the moment we're blocked. Sorry.

**Comment by isaacabraham** [comment]
Linking to #77

**Comment by psfinaki** [comment]
@isaacabraham I see. Sounds quite fatal - and I get the reasoning. Thanks for quick response!

**Comment by isaacabraham** [comment]
@psfinaki hopefully it's not super fatal - it is on the roadmap to fix this :-) You can of course use the TP on full .NET in FSI etc. for exploration still (there's no FSI for .NET Core yet anyway).
